opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -658,7 +658,12 @@ def __exit__(
 
         super().__exit__(exc_type, exc_val, exc_tb)
 
-    def record_exception(self, exception: Exception) -> None:
+    def record_exception(
+        self,
+        exception: Exception,
+        attributes: types.Attributes = None,
+        timestamp: Optional[int] = None,
+    ) -> None:
```

Comment:
the additional parameters here (attributes, timestamp) should be used in the method below. As per the spec:
>If attributes with the same name would be generated by the method already, the additional attributes take precedence

Reply:
@codeboten Can you please guide me on how to do this. I am not able to figure out what exactly needs to be done here.

Reply:
@shreyagupta30 you'll want to pass in the timestamp argument into `add_event`, you can see what it looks like by looking at the method signature here: https://github.com/open-telemetry/opentelemetry-python/blob/937863f8294d6347e131eaaedbc1b877598b13d9/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L589-L593

You'll also want to pass the `attributes` parameter in the `add_event` call below, ensuring that any attributes passed into this method overrides any of the default attributes, the way i'd approach it is probably to create a dict with the default attributes seen below, then loop through the `attributes` arg and set any values in the dict that way.

https://github.com/open-telemetry/opentelemetry-python/blob/937863f8294d6347e131eaaedbc1b877598b13d9/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L697-L701

Last but not least, adding a test like the one below to ensure the attributes are set as expected https://github.com/open-telemetry/opentelemetry-python/blob/937863f8294d6347e131eaaedbc1b877598b13d9/opentelemetry-sdk/tests/trace/test_trace.py#L855-L872